hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileBlock.java

```diff
@@ -1267,6 +1267,8 @@ HFileBlock getBlockForCaching(CacheConfig cacheConf) {
                                 .withCompressTags(fileContext.isCompressTags())
                                 .withIncludesMvcc(fileContext.isIncludesMvcc())
                                 .withIncludesTags(fileContext.isIncludesTags())
+                                .withColumnFamily(fileContext.getColumnFamily())
+                                .withTableName(fileContext.getTableName())
```

Comment:
I see some other paths , which would also use the HFileContextBuilder to build a new HFileContext, should we also attach the cf & tableName in it ?

Reply:
I did so in all the paths I could find where the information was readily available (primarily the main HStore flush path, and the MapReduce output format). It wasn't available in the read path, at least without larger changes.

Reply:
>  It wasn't available in the read path, at least without larger changes.
You mean can not get the tableName & cf in read path ? If so, then  seems it's OK.